### PR TITLE
[DDCNL-7801]re-write journey cache: Update Income

### PR DESCRIPTION
**Pull request by vfryer12**
IncomeController.scala, IncomeSourceSummaryController.scala and JourneyCacheService.scala has been updated to use JourneyCacheNewRepository / UserAnswers and existing journeyCache as a temporary fix, since changing these to use only JourneyCacheNewRepository will make the PR bigger. This will be taken care in [DDCNL-9373](https://jira.tools.tax.service.gov.uk/browse/DDCNL-9373).

- [x] Unit tests passing
- [x] Integration tests passing
- [x] Acceptance tests passing
- [] Reviewed

### CODE REVIEW

File: app/controllers/income/estimatedPay/update/IncomeUpdateCalculatorController.scala [hunk 1]
```diff
@@ -136,52 +128,49 @@ class IncomeUpdateCalculatorController @Inject() (
...
+      val optionalSeq = Seq(
+        userAnswers.get(UpdateIncomeTaxablePayPage),
+        userAnswers.get(UpdateIncomeBonusOvertimeAmountPage),
+        userAnswers.get(UpdateIncomeOtherInDaysPage)
+      )
+
+      (mandatoryJourneyValues, optionalSeq) match {
+        case (mandatory, _) if mandatory.forall(_.isDefined) =>
+          val employer =
+            IncomeSource(id = mandatory(5).getOrElse("").toString.toInt, name = mandatory.head.getOrElse("").toString)
```
Comment (SandeepUmredkar) [inline app/controllers/income/estimatedPay/update/IncomeUpdateCalculatorController.scala:169]: Dont need getOrElse as you have already confirmed it is defined

Comment (askar-makkunnath) [reply app/controllers/income/estimatedPay/update/IncomeUpdateCalculatorController.scala]: Addressed

File: app/uk/gov/hmrc/tai/service/UpdateNextYearsIncomeService.scala [hunk 2]
```diff
@@ -18,32 +18,36 @@ package uk.gov.hmrc.tai.service
...
   employmentService: EmploymentService,
   taxAccountService: TaxAccountService
 )(implicit ec: ExecutionContext) {
 
-  def isEstimatedPayJourneyCompleteForEmployer(id: Int)(implicit hc: HeaderCarrier): Future[Boolean] = {
-    val key = s"${UpdateNextYearsIncomeConstants.Successful}-$id"
-    successfulJourneyCacheService.currentCache.map(_ contains key)
-  }
+  def isEstimatedPayJourneyCompleteForEmployer(id: Int, userAnswers: UserAnswers): Future[Boolean] =
+    journeyCacheNewRepository.get(userAnswers.sessionId, userAnswers.nino).map {
```
[the remark this answers is not in the document]
Comment (askar-makkunnath) [reply app/uk/gov/hmrc/tai/service/UpdateNextYearsIncomeService.scala]: Updated.

File: app/uk/gov/hmrc/tai/service/UpdateNextYearsIncomeService.scala [hunk 3]
```diff
@@ -62,30 +66,43 @@ class UpdateNextYearsIncomeService @Inject() (
...
+    }
+  }
+
+  def getNewAmount(employmentId: Int, userAnswers: UserAnswers): Future[Either[String, Int]] = {
+    val key = UpdateNextYearsIncomeNewAmountPage(employmentId).toString
 
-  def getNewAmount(employmentId: Int)(implicit hc: HeaderCarrier): Future[Either[String, Int]] =
-    journeyCacheService.mandatoryJourneyValueAsInt(amountKey(employmentId))
+    userAnswers.get(UpdateNextYearsIncomeNewAmountPage(employmentId)) match {
+      case Some(value) => Future.successful(Right(value.toInt))
```
[the remark this answers is not in the document]
Comment (askar-makkunnath) [reply app/uk/gov/hmrc/tai/service/UpdateNextYearsIncomeService.scala]: Updated.

**Review by SandeepUmredkar — APPROVED** [review]

**Pull request merged**